javascript/ql/src/semmle/javascript/Variables.qll

```diff
@@ -317,6 +317,24 @@ class LocalVariable extends Variable {
       else result = d.getContainer()
     )
   }
+
+  /**
+   * Gets the location of a declaration of this variable.
+   *
+   * If the variable has multiple declarations, an arbitrary one is used.
+   * If it has no declaration, the location of its declaring scope is used.
+   */
+  Location getLocation() {
+    result =
+      min(Location loc |
+        loc = getADeclaration().getLocation()
+      |
+        loc order by loc.getStartLine(), loc.getStartColumn()
+      )
+    or
+    not exists(getADeclaration()) and
+    result = getDeclaringContainer().getLocation()
```

Comment:
This is in general going to be a big, multi-line location. We could alternatively select the location of the container's CFG start node, which will be a zero-width location at the start of the container.